tdrs-backend/tdpservice/users/api/login.py

```diff
@@ -66,18 +83,25 @@ def validate_and_decode_payload(self, request, state, token_data):
         decoded_payload = self.decode_payload(token_data)
         decoded_id_token = decoded_payload['id_token']
 
+        print(decoded_id_token)
+        print("validate_and_decode_payload:request.session:")
+        print(request.session)
+        for key, value in request.session.items():
+            print('{} => {}'.format(key, value))
+
         if decoded_id_token == {"error": "The token is expired."}:
             raise ExpiredToken("The token is expired.")
 
-        if request.session["state_nonce_tracker"]:
+        if request.session.get("state_nonce_tracker"):
             request.session["token"] = id_token
 
         if not validate_nonce_and_state(request, state, decoded_id_token):
             msg = "Could not validate nonce and state"
             raise SuspiciousOperation(msg)
 
-        if not decoded_id_token["email_verified"]:
-            raise UnverifiedEmail("Unverified email!")
+        # this may not be included in xms
```

Comment:
Still not sure about this, both login.gov and ams use this, but it is an optional one as far as the spec is concerned, so it might not be here @andrew-jameson I have this commented til we know if they want this, It might be related to any sort of flow around XID if there really is a GUID hiding around some where.